### PR TITLE
Add GCP PubSub executor

### PR DESCRIPTION
**Pull request by moshebe**

### CODE REVIEW

File: builtin/bins/dkron-executor-gcppubsub/gcppubsub.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	topic := client.Topic(topicName)
+	res := topic.Publish(ctx, msg)
+	serverID, err := res.Get(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("publish message to topic %q: %w", topicName, err)
+	}
+	return []byte(serverID), nil
+}
+
+func ConfigToPubSubMessage(config map[string]string) (*pubsub.Message, error) {
```
Comment (kostyay) [inline builtin/bins/dkron-executor-gcppubsub/gcppubsub.go:71]: consider making this private
```suggestion
func configToPubSubMessage(config map[string]string) (*pubsub.Message, error) {
```

File: builtin/bins/dkron-executor-gcppubsub/gcppubsub.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		resp.Error = err.Error()
+	}
+	return resp, nil
+}
+
+// ExecuteImpl do http request
+func (g *GCPPubSub) ExecuteImpl(args *dktypes.ExecuteRequest) ([]byte, error) {
+	ctx := context.Background()
+	projectID := args.Config["project"]
+	topicName := args.Config["topic"]
```
Comment (kostyay) [inline builtin/bins/dkron-executor-gcppubsub/gcppubsub.go:39]: consider making the keys const

File: builtin/bins/dkron-executor-gcppubsub/gcppubsub.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	return []byte(serverID), nil
+}
+
+func ConfigToPubSubMessage(config map[string]string) (*pubsub.Message, error) {
+	if config == nil {
+		return nil, fmt.Errorf("invalid config")
+	}
+
+	encodedData := config["data"]
+	attributesJSON := config["attributes"]
```
Comment (kostyay) [inline builtin/bins/dkron-executor-gcppubsub/gcppubsub.go:77]: consider making the keys const

**Review by kostyay — APPROVED** [review]
see comments
LGTM

File: builtin/bins/dkron-executor-gcppubsub/gcppubsub.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	if attributesJSON == "" && encodedData == "" {
+		return nil, fmt.Errorf("at least one of these fields should be set 'attributes, data'")
+	}
+
+	msg := &pubsub.Message{}
+
+	var attributes map[string]string
+	if attributesJSON != "" {
+		err := json.Unmarshal([]byte(attributesJSON), &attributes)
+		if err != nil {
```
Comment (victorcoder) [inline builtin/bins/dkron-executor-gcppubsub/gcppubsub.go:95]:
```suggestion
		if err := json.Unmarshal([]byte(attributesJSON), &attributes); err != nil  {
```

**Review by victorcoder — APPROVED** [review]
LGTM after the nitpicks, thanks for the good work!

**Comment by moshebe** [comment]
Fixed.
Thank you @victorcoder !

**Pull request merged**